src/main.rs

```diff
@@ -11,19 +11,41 @@ use walkdir::WalkDir;
 fn main() {
     // Argument Parsing
     let args: Vec<String> = std::env::args().collect();
+    if args.len() < 2 {
+        eprintln!("Usage: program <folder>");
+        process::exit(1);
+    }
     let folder = PathBuf::from(&args[1]);
 
     // Initialize site data
-    let marmite = fs::read_to_string("marmite.yaml").expect("Unable to read marmite.yaml");
-    let site: Site = serde_yaml::from_str(&marmite).expect("Failed to parse YAML");
+    let marmite = match fs::read_to_string("marmite.yaml") {
+        Ok(data) => data,
+        Err(e) => {
+            eprintln!("Unable to read marmite.yaml: {}", e);
+            process::exit(1);
+        }
+    };
+    let site: Site = match serde_yaml::from_str(&marmite) {
+        Ok(site) => site,
+        Err(e) => {
+            eprintln!("Failed to parse YAML: {}", e);
+            process::exit(1);
+        }
+    };
     let mut site_data = SiteData::new(&site);
 
     // Walk through the content directory
     for entry in WalkDir::new(folder.join(site_data.site.content_path)) {
```

Comment:
Switching to `match` when iterating over `WalkDir` makes the code more robust and safer when handling errors. Instead of using `unwrap()`, which could cause a panic if there's an issue reading an entry, `match` lets us handle each error individually, log it, and move on to the next entry.